plugins/inputs/prometheus/prometheus_test.go

```diff
@@ -137,6 +138,36 @@ func TestPrometheusGeneratesMetricsAlthoughFirstDNSFailsIntegration(t *testing.T
 	require.True(t, acc.HasTimestamp("test_metric", time.Unix(1490802350, 0)))
 }
 
+func TestPrometheusGeneratesMetricsSlowEndpoint(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(4 * time.Second)
+		_, err := fmt.Fprintln(w, sampleTextFormat)
+		require.NoError(t, err)
+	}))
+	defer ts.Close()
+
+	p := &Prometheus{
+		Log:             testutil.Logger{},
+		URLs:            []string{ts.URL},
+		URLTag:          "url",
+		ResponseTimeout: config.Duration(time.Second * 5),
```

Comment:
Thanks for the test, can you put up a test that causes the timeout to get hit as well please?

Reply:
Sure. I'll do that.